v3/app/controllers/api/v3/roles_controller.rb

```diff
@@ -0,0 +1,24 @@
+class Api::V3::RolesController < ApplicationController
+
+  def get_roles
+    ip = request.remote_ip
+    token = params[:token].to_s.strip.downcase
+
+    status, user = User.validate_token({:ip => ip, :token => token})
```

Comment:
The token HAS to be validated on EACH call! There is no persistent connection to the DB so the token is the initial gate on every API call.  I will see if I can pull it out as a method and process it in a :before_action 

Also, it is actually IMPORTANT that validate_token returns a user.  (a) the front-end can verify that the token belongs to the user to whom they expect it to (that is not particularly critical), and (b) the front-end can get the permissions assigned to this token/user from the response, so with a simple "vaidate_token" API call can determine what items to show/not show in the GUI WITHOUT having to make a second call the to server.

Reply:
my point is that if I read the name `validate_token`,  I'm not expecting the method to return a user.

Reply:
something like "token_get_user"

Reply:
DONE. implemented "check_token_for_permission()" call which takes an optional permission_id. It returns either a status + error <or> a status + user.  I used "check_token" instead of "get_user_from_token" because at the end of the day you are checking the token (which technically IS a user), and the "for_permission" describes the optional parameter.